File: basics_of_computer_algorithms/chapter_one/TASK1_1/pascal/TASK1_1.pas
program EX1_1;
uses Crt;
var x_1, x_2, p, q, temp_p, temp_q: real;
begin
    read(temp_p);
    if temp_p = 0 then
    begin
        exit;
    end;
    
    read(temp_q);
    if temp_q = 0 then
    begin
        exit;
    end;

    p := temp_p;
    q := temp_q;

    x_1 := (-p - Sqrt( Sqr(p) - 4*q))/2;
    x_2 := (-p + Sqrt( Sqr(p) - 4*q))/2;

    writeln(x_1);
    writeln(x_2);

    Delay(3000);
end.

File: basics_of_computer_algorithms/chapter_one/TASK1_1/cpp/TASK1_1.cpp
#include <iostream>

int main()
{
    double x_1, x_2, p, q, temp_p, temp_q;

    std::cin >> p;
    if (p == 0) exit(1);
    std::cout <<std::endl;

    std::cin >> q;
    if (q == 0) exit(1);
    std::cout <<std::endl;

    x_1 = (-p - sqrt( p*p - 4*q))/2;
    x_2 = (-p + sqrt( p*p - 4*q))/2;

    std::cout << x_1 << std::endl;
    std::cout << x_2 << std::endl;

    return 0;
}